cmd/cli/proxy.go

```diff
@@ -1,17 +1,37 @@
 package main
 
 import (
+	"fmt"
 	"io"
+	"os"
 
 	"github.com/spf13/cobra"
 	"helm.sh/helm/v3/pkg/action"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/openservicemesh/osm/pkg/cli"
+	"github.com/openservicemesh/osm/pkg/constants"
 )
 
 const proxyCmdDescription = `
 This command consists of subcommands related to the operations
 of the sidecar proxy on pods.
 `
 
+type proxyAdminCmd struct {
+	out        io.Writer
+	config     *rest.Config
+	clientSet  kubernetes.Interface
+	query      string
+	namespace  string
+	pod        string
```

Comment:
I'm hoping to push more stuff behind interfaces and unify behind a single client (k8s.Client for k8s, and kube.Client will abstract that)

I wonder if we can map things like pod, to proxy, and have a generic ProxyPortForwarder which abstracts the k8s one.

The rest of the client setup isn't quite there yet (a few more PR's, and need to get the k8s client to be able to issue updates, and have options for queries outside the informer cache), but if it's feasible to add some abstractions in this PR I'd push for it. 

If not, no worries.

Reply:
I am not sure I understand what that would achieve. The CLI code is heavily decoupled from internal control plane infra clients, and relies on kubeclient directly. More so, I don't believe we should refactor existing CLI behavior in this change, the purpose of which is to extend the existing `osm proxy` command.

If there isn't an issue for your proposal open already, could you open one and describe how you'd like to address the existing CLI code that uses the kubeclient?